PerfTools/AllocMonitor/plugins/ModuleEventAllocMonitor.cc

```diff
@@ -0,0 +1,445 @@
+        ++streamSync_[iStream.streamID().value()];
+      }
+    });
+
+    iAR.watchPreModuleEventAcquire([this](auto const& iStream, auto const& iMod) {
+      auto index = moduleIndex(module_id(iMod));
+      streamModuleInAcquire_[nModules_ * iStream.streamID().value() + index].store(true);
+      filter_.startOnThread(module_id(iMod));
+    });
+    iAR.watchPostModuleEventAcquire([this, file](auto const& iStream, auto const& iMod) {
+      auto mod_id = module_id(iMod);
+      auto info = filter_.stopOnThread(mod_id);
+      if (info) {
+        assert(info->allocMap_.allocationSizes().size() == info->allocMap_.size());
+        auto v = std::accumulate(info->allocMap_.allocationSizes().begin(), info->allocMap_.allocationSizes().end(), 0);
+        std::stringstream s;
+        s << "A " << mod_id << " " << iStream.streamID().value() << " " << info->totalMatchedDeallocSize_ << " "
+          << info->numMatchedDeallocs_ << " " << info->totalUnmatchedDealloc_ << " " << info->numUnmatchedDeallocs_
+          << " " << v << " " << info->allocMap_.allocationSizes().size() << "\n";
+        file->write(s.str());
+        //edm::LogSystem("MEA")<< "temp mem size "<<info->totalMatchedDeallocSize_ <<" # "<<info->numMatchedDeallocs_ <<"\nold mem size "<<info->totalUnmatchedDealloc_<<" # "<<info->numUnmatchedDeallocs_ <<
+        //  "\nnew mem size "<< v <<" # "<<info->allocMap_.allocationSizes().size();
+        auto index = mod_id;
+        if (not moduleIDs_.empty()) {
+          auto it = std::lower_bound(moduleIDs_.begin(), moduleIDs_.end(), mod_id);
+          index = it - moduleIDs_.begin();
+        }
+        {
+          auto const& alloc = streamModuleAllocs_[nModules_ * iStream.streamID().value() + index];
+          assert(alloc.size() == alloc.allocationSizes().size());
+        }
+        streamModuleAllocs_[nModules_ * iStream.streamID().value() + index] = info->allocMap_;
+        {
+          auto const& alloc = streamModuleAllocs_[nModules_ * iStream.streamID().value() + index];
+          assert(alloc.size() == alloc.allocationSizes().size());
+        }
+        ++streamSync_[iStream.streamID().value()];
+        streamModuleInAcquire_[nModules_ * iStream.streamID().value() + index].store(false);
+      }
+    });
+    /*
+    iRegistry.watchPreModuleEventDelayedGet(
+                                            StreamEDModuleState<Step::preModuleEventDelayedGet>(logFile, beginTime, iFilter));
+    iRegistry.watchPostModuleEventDelayedGet(
+                                             StreamEDModuleState<Step::postModuleEventDelayedGet>(logFile, beginTime, iFilter));
+    iRegistry.watchPreEventReadFromSource(
+                                          StreamEDModuleState<Step::preEventReadFromSource>(logFile, beginTime, iFilter));
+    iRegistry.watchPostEventReadFromSource(
+                                           StreamEDModuleState<Step::postEventReadFromSource>(logFile, beginTime, iFilter));
+    */
```

Comment:
What is the purpose of these commented out registration calls?

Reply:
I was worried that those might have to be implemented to avoid assigning Source related memory calls to the module. Still haven't determined if that is needed yet.

Reply:
Could you add a comment about the need of those callbacks being unclear at the moment?

Reply:
Added comment to the code.